src/coreComponents/constitutive/contact/ContactBase.cpp

```diff
@@ -29,16 +31,122 @@ namespace constitutive
 ContactBase::ContactBase( string const & name,
                           Group * const parent ):
   ConstitutiveBase( name, parent ),
-  m_penaltyStiffness( 0.0 )
+  m_penaltyStiffness( 0.0 ),
+  m_shearStiffness( 0.0 ),
```

Comment:
Do we want to generalize this class for non-penalty approaches in the future?
The name `ContactBase` is very general ...

Reply:
I agree we should maybe have a hierarchy `ContactBase->PenaltyBase->SomePenaltyModel` to move these specific member variables somewhere more appropriate.

Reply:
I agree with @CusiniM . I would like to se the constitutive relation be just that. The constraint enforcement method should indeed be a separate/PhysicsSolver type thing.